packages/relay/src/lib/services/hbarLimitService/index.ts

```diff
@@ -375,12 +421,12 @@ export class HbarLimitService implements IHbarLimitService {
 
   /**
    * Creates a basic spending plan for the given eth address.
-   * @param {string} ethAddress - The eth address to create the spending plan for.
+   * @param {string} [ethAddress] - The eth address to create the spending plan for.
    * @param {string} [ipAddress] - The ip address to create the spending plan for.
    * @returns {Promise<IDetailedHbarSpendingPlan>} - A promise that resolves with the created spending plan.
    * @private
    */
-  private async createBasicSpendingPlan(ethAddress: string, ipAddress?: string): Promise<IDetailedHbarSpendingPlan> {
+  private async createBasicSpendingPlan(ethAddress?: string, ipAddress?: string): Promise<IDetailedHbarSpendingPlan> {
```

Comment:
This change has altered the validity of the `ethAddress`, as it can potentially be read as null in cases where it is used for helper queries that do not necessarily involve `originalCallerAddress`, as explained in the PR's description.

The fallback logic for cases where neither `ethAddress` nor `ipAddress` is valid is handled within the first block of the method:

```typescript
if (!ethAddress && !ipAddress) {
      throw new Error('Cannot create a spending plan without an associated eth address or ip address');
}
```

Reply:
Still, we don’t ever expect `createBasicPlan` to be called in those situations, if no IP/ETH address is passed to `shouldLimit` we want to check only against the `totalLimit` of the HBAR limiter and we don’t include the logic with the spending plans. Creating an empty spending plan without an IP/ETH address linked to it would only fill the memory in cache with unnecessary entries which will be used only once (for the request in which they were created) and don’t really provide any additional value.

Please, revert this change.

Reply:
https://github.com/hashgraph/hedera-json-rpc-relay/pull/3024#discussion_r1786784941